Add config option to stop chunkgen when above worldheight

ChimneySwift opened the pull request:
Config option to help reduce disk usage where players try to fly excessively far, often doing so above y=256

ChimneySwift committed:
Add setting to stop chunkgen when above worldheight

Proximyst reviewed (commented):
What happens if you enable the option to collide on unloaded chunks?

ChimneySwift committed:
Merge remote-tracking branch 'upstream/master'

ChimneySwift committed:
Update patch name

ChimneySwift committed:
Fix compat with prevent movement into unloaded chunks

ChimneySwift committed:
Stop entities loading chunks when running checks

This patch resolves #4805 and should result in performance improvements when players move into unloaded chunks.

Main changes in this commit:

- Ensure EntityLiving.g (travel) doesn't load chunks
- Ensure EntityLiving.isClimbing doesn't load chunks
- Ensure Entity.isInBubbleColumn only checks loaded chunks
- Don't attempt to spawn sprint particles if flying
- Avoid loading chunks when performing movement checks
- Avoid Entity.checkBlockCollisions loading chunks
- Only update fluid in loaded chunks
- Ensure chunkCheck doesn't load chunks for players that shouldn't
- Don't perform collision check when player is spectator or outside world
- Fix movement check causing chunk load for players who can't load chunks
- Don't check fall damage in unloaded chunks

These changes have not been extensively ingame-tested thus I cannot confirm they do not have side-effects, only that they resolve the issue.

kashike commented on Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch:
```diff
@@ -0,0 +1,299 @@
+                         blockposition_mutableblockposition.d(i, j, k);
+-                        IBlockData iblockdata = this.world.getType(blockposition_mutableblockposition);
+ 
+-                        try {
```
Please avoid changing the indentation of the existing code here.

ChimneySwift replied:
I might be missing something here, but the indentation here is due to an added if statement. It looks like the comments are in the wrong spot however, which might be the actual issue here.

electronicboy replied:
min-diff, changing the indentation causes more diff noise than is needed, causing un-needed code conflicts when merging in upstream changes, etc, makes the cost of keeping it pretty too high to be worth it

ChimneySwift replied:
Ah I see what you mean.

ChimneySwift replied:
Believe latest commit should resolve this

ChimneySwift committed:
Avoid excess indentation

Also fixed style for isOutsideWorld and moved comments to more logical locations.

ChimneySwift committed:
Fix broken logic

Not sure what I was thinking here.

Curiously this didn't actually result in chunks generating. Still good for performance however.
